<launch>
  <node name="ps3joy" pkg="ps3joy" type="ps3joy.py" output="screen" respawn="true" args="--continuous-output" />
  
  <node pkg="joy" type="joy_node" name="ps3_joy" output="screen" >
    <param name="dev" type="string" value="/dev/input/js0" />
    <param name="deadzone" value="0.12" />
  </node>

  <node pkg="diagnostic_aggregator" type="aggregator_node" name="diagnostic_aggregator" >
    <!-- Load the file you made above -->
    <rosparam command="load" file="$(find ps3joy)/diagnostics.yaml" />
  </node>
  
  	<node name="base" pkg="base" type="base" output="screen" respawn="false">
		<!-- launch-prefix="gdb -ex run - -args" -->
		<param name="motor_port" value="motor_comm"/>
		<param name="motor_config" value="$(find base)/cfg/wheels.xml"/>
		<param name="wheel_diameter" value="0.295"/>
		<param name="wheel_base" value="0.54"/>
	</node>

	<node name="motor_comm" pkg="shared_serial" type="server" output="screen">
		<param name="port_name" value="/dev/ttyUSB0"/>
		<param name="port_type" value="RS485_FTDI"/>
		<param name="baud_rate" value="921600"/>
	</node>

</launch>
